### ISSUE TITLE
It's not possible to filter user's sentences by given language

### ISSUE THREAD

**Issue by Guybrush88** [Tatoeba/tatoeba2#2630]
**To Reproduce**
1. Go to a user's page listing all his / her sentences (for example: https://tatoeba.org/ita/sentences/of_user/Guybrush88)
2. Try to select a given language (for example French)
3. See the maintenance page after a while
*Example: *
*1. Go here: https://tatoeba.org/ita/sentences/of_user/Guybrush88
*2. Select French as the given language
*3. Wait for the new page to be loaded
*4. Receive the maintenance page*

**Expected behavior**
One should be able to succesfully see all the user's sentences in a given language.

**Comment by alanfgh** [comment]
I see the same behavior when I follow these steps.

**Comment by jiru** [comment]
It’s because you have too much sentences @Guybrush88 :smile: The problem doesn’t happen if you browse other members’ sentences. The database query takes too long and it times out. We need to optimize that query.

**Comment by Guybrush88** [comment]
@jiru thanks for the very quick fix :D I had checked also with my profile in order to post a bug report, but I discovered it with other users' profiles (I think I was trying to browse some English sentences by @alanfgh to translate them into Italian)

**Comment by alanfgh** [comment]
Glad to hear that you wanted to translate my sentences, @Guybrush88 . :) I can reproduce the same problem sporadically (on the production site) when searching through my own sentences. Interestingly enough, it happened not when I searched for English sentences (of which I own 14,000+), but the first time when I searched for Esperanto sentences (of which I own only 35). The second time I searched for Esperanto sentences, there was no error, but it did take a while.
